From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Loki Rautio <lokirautio@gmail.com>
Date: Sat, 7 Oct 2023 17:25:13 -0500
Subject: [PATCH] always canSee for now


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index f70bd402f6d8249c445fd99a52feb344f7f24dcb..5860f3c26c76c5dd903c3797ee5a48bf04380036 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2080,7 +2080,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public boolean canSee(org.bukkit.entity.Entity entity) {
-        return this.equals(entity) || entity.isVisibleByDefault(); // SPIGOT-7312: Can always see self
+        return true; // SPIGOT-7312: Can always see self
     }
 
     public boolean canSee(UUID uuid) {
