From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Loki Rautio <lokirautio@gmail.com>
Date: Sat, 7 Oct 2023 17:25:36 -0500
Subject: [PATCH] revert to 1.19.2-like updatePlayer method call

missing optimization in 1.20.2 for some reason

diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 0c2617574e21037d94ac56ad08b490f9bca5c5af..97efbc53e172de7dff17dfcfbb0723c203107053 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1285,20 +1285,23 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         public void updatePlayer(ServerPlayer player) {
             org.spigotmc.AsyncCatcher.catchOp("player tracker update"); // Spigot
             if (player != this.entity) {
-                Vec3 vec3d = player.position().subtract(this.entity.position());
-                int i = ChunkMap.this.getPlayerViewDistance(player);
-                double d0 = (double) Math.min(this.getEffectiveRange(), i * 16);
-                double d1 = vec3d.x * vec3d.x + vec3d.z * vec3d.z;
+                // Paper start - remove allocation of Vec3D here
+                // Vec3 vec3d = player.position().subtract(this.entity.position());
+                double vec3d_dx = player.getX() - this.entity.getX();
+                double vec3d_dz = player.getZ() - this.entity.getZ();
+                // Paper end - remove allocation of Vec3D here
+                double d0 = (double) Math.min(this.getEffectiveRange(), io.papermc.paper.chunk.system.ChunkSystem.getSendViewDistance(player) * 16); // Paper - per player view distance
+                double d1 = vec3d_dx * vec3d_dx + vec3d_dz * vec3d_dz; // Paper
                 double d2 = d0 * d0;
-                boolean flag = d1 <= d2 && this.entity.broadcastToPlayer(player) && ChunkMap.this.isChunkTracked(player, this.entity.chunkPosition().x, this.entity.chunkPosition().z);
+                boolean flag = d1 <= d2 && this.entity.broadcastToPlayer(player);
                 // Paper start - check Y
-                if (flag && level.paperConfig().entities.trackingRangeY.enabled) {
-                    double rangeY = level.paperConfig().entities.trackingRangeY.get(this.entity, -1);
-                    if (rangeY != -1) {
-                        double vec3d_dy = player.getY() - this.entity.getY();
-                        flag = vec3d_dy * vec3d_dy <= rangeY * rangeY;
-                    }
-                }
+//                if (flag && level.paperConfig().entities.trackingRangeY.enabled) {
+//                    double rangeY = level.paperConfig().entities.trackingRangeY.get(this.entity, -1);
+//                    if (rangeY != -1) {
+//                        double vec3d_dy = player.getY() - this.entity.getY();
+//                        flag = vec3d_dy * vec3d_dy <= rangeY * rangeY;
+//                    }
+//                }
                 // Paper end - check Y
 
                 // CraftBukkit start - respect vanish API
